### DISCUSSION TITLE
Cannot Enforce HTTPS

### DISCUSSION THREAD

**Discussion by amanuel2** [community/community#22708] [Other Feature Feedback, Questions, & Ideas]
<p>I have registered my domain <a href="http://amanuel.io" rel="nofollow noopener">http://amanuel.io</a> via namecheap and am hosting it on github : <a href="https://github.com/amanuel2/amanuel2.github.io" rel="nofollow noopener">https://github.com/amanuel2/amanuel2.github.io</a> .</p>
<p>When i try to enforce https its greyed out saying: Enforce HTTPS — Unavailable for your site because your domain is not properly configured to support HTTPS ( — Troubleshooting custom domains</p>
<p>Any help would be greatly appreciated</p>
<h2>EDIT</h2>
<p>Here is my current records for this website<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/a/a031425e00e6672249487c645e0ad6a40e9b999a.png" data-download-href="/uploads/short-url/mR7VPnF5CAPdubeyGwETFrCLoNk.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/16020076/181072661-82fa255e-8a09-4ab5-9567-a7d58582409a.png" alt="image" data-base62-sha1="mR7VPnF5CAPdubeyGwETFrCLoNk" width="690" height="161" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/a/a031425e00e6672249487c645e0ad6a40e9b999a_2_690x161.png, https://aws1.discourse-cdn.com/github/optimized/2X/a/a031425e00e6672249487c645e0ad6a40e9b999a_2_1035x241.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/a/a031425e00e6672249487c645e0ad6a40e9b999a_2_1380x322.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/a/a031425e00e6672249487c645e0ad6a40e9b999a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2799×655 10.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by amanuel2** [comment]
<aside class="quote no-group" data-username="Outdoorballpro" data-post="2" data-topic="127468" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 Outdoorballpro:</div>
<blockquote>
<p>Please advise if you are satisfied being a current user of github hosting then I shall move forward to transfer hosting.</p>
</blockquote>
</aside>
<p>Im confused. is this a troll account?</p>

**Comment by tcbyrd** [comment, marked as answer]
<p><a class="mention" href="/amanuel2">@amanuel2</a> The IPs you’re using for the A records are pointed to our old IPs, which don’t go through the same endpoints that allow us to do automatic HTTPS. Docs on this are available here: <a href="https://docs.github.com/en/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain" rel="nofollow noopener">https://docs.github.com/en/github/working-with-github-pages/managing-a-custom-domain-for-your-github-pages-site#configuring-an-apex-domain</a></p>
<p>Once you point the site’s A records to the IPs listed in that doc, we should be able to obtain a certificate for you. Once the certificate is obtained, the option to enforce HTTPS should become available.</p>

**Reply by danhislop** [reply]
the problem for me was that there was an existing www record for my DNS host. i followed these instructions from github to fix it: "Navigate to your DNS provider and create a CNAME record for the www subdomain that points to your GitHub Pages default domain. For example, if your site is located at <user>.github.io, you should create a CNAME record that points www.example.com to <user>.github.io Similarly, for an organization site located at <organization>.github.io, you should create a CNAME record that points www.example.com to <organization>.github.io. Ensure that the CNAME record points directly to <user>.github.io or <organization>.github.io without including the repository name."

**Comment by wwwnmr** [comment]
![1fc635f5-b2b6-440b-83dd-765a1383486d.png](https://github.com/community/community/assets/113519777/cccd8809-8871-4f3a-9ac5-94e0b6c1e96e)

**Comment by wwwnmr** [comment]
![1fc635f5-b2b6-440b-83dd-765a1383486d.png](https://github.com/community/community/assets/113519777/8f777b49-e6c7-4d00-b2de-5a0d0f8b82fe)
@wwwnmr
